Lower Required Python Version from 3.10 to 3.8

KPreisner opened the pull request:
This PR lowers the required python version from `3.10` to `3.8` so users can more accessibly install the `sindri` PyPI module. Many users have a python version lower than `3.10` on their systems by default.

closes #42 

The only production dependency for this repo is the `requests` module, which [requires](https://pypi.org/project/requests/) `>=3.8`.

Note: The `dev` poetry group requires `3.8.1` because that is the minumum for [flake8](https://pypi.org/project/flake8/).

KPreisner commented:
Question for the reviewer:

Should we set the minimum to 3.9? 3.8 has reached its end of life for security updates according to [this](https://devguide.python.org/versions/).

grahamu commented:
> Many users have a python version lower than 3.10 on their systems by default.

How do we know this? Is this a problem for our users? What versions are they using?

> Should we set the minimum to 3.9? 3.8 has reached its end of life for security updates according to https://devguide.python.org/versions/.

Should we be forcing users to upgrade if they're on 3.8x? Or are we providing a service which works with older Python versions regardless of support/EOL?

grahamu commented:
I support this SDK working with Python 3.8. However, since Python 3.8 is no longer supported by the PSF, I suggest we add verbiage to README which explains how the SDK works with Python version 3.8 but we don't officially support that usage. Further please show which Python versions we do officially support with our SDK.

grahamu reviewed (commented):
Please drop the Python requirement to 3.8 from 3.8.1.
Please update the README to indicate our supported/unsupported Python versions.

KPreisner commented:
I updated the Python requirement [8c5b442](https://github.com/Sindri-Labs/sindri-python/pull/59/commits/8c5b442a28c66be3910a0f85ea3088d7834080d2) so it is 3.8 and 3.8.1 for the dev dependencies. I also had to update the typing annotations [098fa19](https://github.com/Sindri-Labs/sindri-python/pull/59/commits/098fa196f8cc482b60b9b21114f410414e91466c) in the code to use the old typing syntax so the code can run with 3.8.0. 

I added more clarity to the README [5bb9d4c](https://github.com/Sindri-Labs/sindri-python/pull/59/commits/5bb9d4cded5e41ac683a3abb6fab4e2a253ffa03) as well.

I ran the included `pytest` tests with a python version of `3.8.0` and they all pass now with the above updates.

grahamu commented on README.md:
```diff
@@ -10,6 +10,14 @@
 Contains SDK code for Sindri APIs. Please see [sindri.app](https://sindri.app) for more details.
 
+## Requirements
+This is compatible with Python `>=3.8`.
```
```suggestion
This SDK is compatible with Python `>=3.8`.
```

grahamu commented on pyproject.toml:
```diff
@@ -38,16 +38,17 @@ repository = "https://github.com/Sindri-Labs/sindri-python"
 requests = "^2.13.0"
 
 [tool.poetry.group.dev.dependencies]
+python = ">=3.8.1"
```
I don't know about this dependency group. Is >=3.8.1 correct when we allow 3.8 minimum elsewhere?

KPreisner replied:
This allows us to do `poetry install --no-dev` to install just the core dependency group when running in python `3.8.0`. But if you want to install the `dev` dependency group, you'll need `3.8.1` because `flake8` requires `3.8.1`.

grahamu replied:
Perfect, thanks!

grahamu reviewed (approved):
LGTM thanks for backsliding down to 3.8!
